Ignore Friend Recommendations Feature

DanielNosrati opened the pull request:
Implements the feature to ignore certain users from your recommended Facebook friends.
Also fixes a leak where we created FB test users and didn't delete them.

DanielNosrati committed:
Ignore Friend Recommendations Feature

DanielNosrati commented on src/app/pcasts/dao/users_dao.py:
```diff
@@ -163,3 +163,21 @@ def search_facebook_users(facebook_ids, user_id, offset, max_search \
+    return fb_friend_ids
+  else:
+    ignored_ids = [id for id in ignored_ids.ignored_fb_ids.split(',')]
+    return list(set(fb_friend_ids) - set(ignored_ids))
```
Not sure if this is deterministic by calling set?

kevingreer replied:
I don't think it guarantees a certain order. Could do something like ` a = set(ignored_ids)` and `[item for item in fb_friend_ids if item not in a]`

DanielNosrati commented on src/app/pcasts/models/ignored_users.py:
```diff
@@ -0,0 +1,18 @@
+
+  user_id = db.Column(db.Integer, \
+      db.ForeignKey('users.id', ondelete='CASCADE'), primary_key=True)
+  ignored_fb_ids = db.Column(db.Text)
```
I'll run some benchmarks/ask someone more knowledgeable about this.
I would imagine there aren't that many people you want to ignore so a pure relational way of storing
user_id, ignored_id might be more optimal than the string.

kevingreer replied:
If we're adding a string for each user, why wouldn't we add the `ignored_fb_ids` to the user model instead?

DanielNosrati replied:
Part of it was wanting to avoid the insert anomaly of having too many nulls in the db and another part of it was thinking that the relational way might be better than just keeping strings in the db

kevingreer commented on src/scripts/test_podcast.py:
```diff
@@ -2,15 +2,18 @@
 
 SESSION_TOKEN = 0
-SERVER_URL = "http://SERVER_URL/api/v1/search/users/A/?offset=0&max=50"
+SERVER_URL = 'localhost:5000'
```
will we always be on port 5000?

DanielNosrati replied:
Just a dummy value. We can put in the real destination in case we ever want to test against production

kevingreer commented on src/app/pcasts/dao/users_dao.py:
```diff
@@ -163,3 +163,21 @@ def search_facebook_users(facebook_ids, user_id, offset, max_search \
+    ignored_ids.ignored_fb_ids = ignored_ids.ignored_fb_ids + \
+        ",{}".format(to_ignore_id)
+  db_utils.commit_model(ignored_ids)
+  return True
```
i don't see the point of returning True since we never return False, if we're trying to see if an exception was thrown there's already a `success` parameter in the default response of an `AppDevController` that has that info

DanielNosrati replied:
Good point. I initially didn't want to send an empty response but i guess that is fine

DanielNosrati committed:
PR Feedback

Added empty response to endpoint, changed ignored filtering logic and updated tests

DanielNosrati committed:
Merge branch 'master' into daniel/FacebookIgnore
